### DISCUSSION TITLE
sync problem with ios

### DISCUSSION THREAD

**Discussion by stoi2000** [ckulka/baikal-docker#58] [Q&A]
Hi
i have just installed baikal on raspberry pi and it works fine in browser, evolution mail on desktop and even under blackberry os10. thx for that! If i am trying to sync my baikal calendar with my iphone (ios 13.5.1) the sync will not be sucessfull. I have found some informations that ios is not properly working because there are some rewriting rules inside the webserver are missing.

thanks for help in advance!
stoi2000

**Comment by ckulka** [comment]
Hi there, can you link the articles you found? I'll have a look and can add those changes.

**Comment by stoi2000** [comment]
Hi ckulka
Thanks for your reply and sorry for my late answer. The article i have found is in german language and can be found under: https://www.cyon.ch/blog/CalDAV-CardDAV-Server-Baikal#Setup
I'll translate the important Information into ENG:
![baikal](https://user-images.githubusercontent.com/67918401/87054114-099a0280-c203-11ea-9d07-bbd50992b62c.png)
`for macOS, iOS and iPadOS Users
When the accounts set up via Baïkal are set up later, the Apple operating systems search for the corresponding PHP files in the wrong place. In this case the following entry in the control file .htaccess, which must be located one directory above the Baïkal subdomain, helps:

RewriteEngine On
RedirectMatch 301 ^/.well-known/caldav /baikal/html/cal.php
RedirectMatch 301 ^/.well-known/carddav /baikal/html/card.php
`
I am really new to docker so it's maybe on my side that i cannot establish a connection to my iphone.
Thanks for your support and help!

stoi2000

**Comment by ckulka** [comment]
Hi @stoi2000,

thanks, unfortunately I don't think that's it because it is already part of the Docker image (both apache and nginx):

https://github.com/ckulka/baikal-docker/blob/a8e7d1159250d61da6c532c72bc475921e991757/files/apache.conf#L9-L11

https://github.com/ckulka/baikal-docker/blob/a8e7d1159250d61da6c532c72bc475921e991757/files/nginx.conf#L10-L11

I used the configuration setup from the official installation and troubleshooting guide. Difference to your article is that both are served from dav.php, I believe they merged cal.php and card.php in the 0.6 release... I think.

* <https://sabre.io/baikal/troubleshooting/>
* <https://sabre.io/baikal/install/#apache-vhost-installation>

Can you test if that redirect works on your setup by going to <https://your-domain.com/.well-known/carddav> and check if you are being redirected to <https://your-domain.com/dav.php>?

**Comment by BobWs** [comment]
I have also experienced sync problems. I am using Baikal behind a reverse proxy and found out that if you redirect to port 80 then `.well-known` does not work, but if I redirect to port 443 then `.well-known` works. And this without changing the `.htacces` File.


![F90EC85D-CB81-4A0E-A308-33E34EFA01B5](https://user-images.githubusercontent.com/5485820/111063395-837b3d80-84ae-11eb-83b9-e7fbdfa47d80.jpeg)


![545A50E2-0C66-4378-B46A-F638BAB5EA64](https://user-images.githubusercontent.com/5485820/111063396-8413d400-84ae-11eb-92b4-1f2eeb0787a0.jpeg)

**Comment by ckulka** [comment]
Hi @stoi2000 ,

I'm guessing you solved this issue already somehow, but in case you haven't or someone else comes across this: check out #37, someone else had an issue with iOS when using Baikal with Traefik.

The fix is to add rewrite rules to Traefik (unsure why it's needed there too...) and the example has been updated:

https://github.com/ckulka/baikal-docker/blob/8cb1ae30b3374a050e029e84cd99dadfdf234b5c/examples/docker-compose.ssl.yaml#L18-L22

**Comment by kvndl** [comment]
Are there any current fixes for getting iOS operational without the use of Traefik? Currently have everything going through my own Nginx setup using [NPM](https://github.com/jc21/nginx-proxy-manager).

Attempted to add those rewrite rules manually, but the error I get when trying to sync remains the same: `Unable to connect to account` -- this is after signing into said account via Settings.

When I try to reload/sync my calendar, it hits me with the following logs: [Gist](https://gist.github.com/sch3p/a7456fa5cdde41f5911a20467792d1a5)

Worth mentioning that everything works without a hitch when using a client such as Evolution.

Thanks in advance for any intel! 😄

**Reply by ckulka** [reply]
Sorry, I missed this questions completely. Can you try the following: change the target port of the Baikal container in NPM from port 80 to 443 and recreate the account in iOS?

My guess is that Apache first redirects on the plain HTTP port 80 instead of the HTTPS port 443, because (from Apache's perspective) the request came on port 80. By routing the incoming request to Baikal's HTTPS port 443, the redirect remains on HTTPS and it should be fine.

This is basically the same solution proposed by @BobWs just for NPM, with my explanation as to why it works.

**Reply by Jezz3** [reply]
I had the same Problem. Tested it and now it works. For the URL I use https://example.domain.com/dav.php/principals/myuser
Thank you very much for the solution.
